qiskit/compiler/transpile.py

```diff
@@ -631,14 +631,17 @@ def _parse_scheduling_method(scheduling_method, num_circuits):
     return scheduling_method
 
 
-def _parse_instruction_durations(backend, inst_durations, dt, scheduling_method, num_circuits):
+def _parse_instruction_durations(backend, inst_durations, dt, num_circuits):
     durations = None
-    if scheduling_method is not None:
-        from qiskit.transpiler.instruction_durations import InstructionDurations
-        if backend:
-            durations = InstructionDurations.from_backend(backend).update(inst_durations, dt)
-        else:
-            durations = InstructionDurations(inst_durations, dt)
+    if inst_durations is None and backend:
+        try:
+            backend_durations = InstructionDurations.from_backend(backend)
+        except AttributeError:
+            backend_durations = InstructionDurations()
+        durations = backend_durations.update(inst_durations, dt)
```

Comment:
I see this line is still necessary to update `dt`. `durations = backend_durations.update(None, dt)` might be clearer.